diff --git a/src/agenda/AgendaEventRenderer.js b/src/agenda/AgendaEventRenderer.js
index 5842ed4..38c8dba 100644
--- a/src/agenda/AgendaEventRenderer.js
+++ b/src/agenda/AgendaEventRenderer.js
@@ -653,6 +653,7 @@ function AgendaEventRenderer() {
             "<span class='event-title-txt'>" + htmlEscape(event.title) + " </span>" +
             "</div>" +
             "</div>" +
+            // ( event.iconUrl ? "<span class='event-icon-url' style='background-image:url(" + event.iconUrl + ")'> </span>" : "" ) +
             "</div>" + // close inner
 			"<span class='event-badges'></span>";
         if (seg.isEnd && isEventResizable(event)) {
@@ -740,12 +741,14 @@ function AgendaEventRenderer() {
         var revert;
         var allDay=true;
         var dayDelta;
+        var origColDragStart;
         var dis = opt('isRTL') ? -1 : 1;
         var hoverListener = getHoverListener();
         var colWidth = getColWidth();
         var slotHeight = getSlotHeight();
         var minMinute = getMinMinute();
         var helperElement;
+        var _ui;
         eventElement.draggable({
             helper: 'clone',
             appendTo: 'body',
@@ -760,9 +763,13 @@ function AgendaEventRenderer() {
                 origHeight = eventElement.height();
                 eventElement.css('display', 'none');
                 helperElement = ui.helper;
+                _ui = ui;
                 // As the helper is no longer in the all-day bar, it becomes higher if not implicitly set
                 helperElement.css('height', origHeight);
                 hoverListener.start(function(cell, origCell, rowDelta, colDelta) {
+                    if (!origColDragStart) {
+                        origColDragStart = cell.col;
+                    }
                     clearOverlays();
                     if (cell) {
                         //setOverflowHidden(true);
@@ -787,6 +794,12 @@ function AgendaEventRenderer() {
                                             (event.end && event.end.getHours() !== 0 ? ((event.end - event.start) / MINUTE_MS) : opt('defaultEventMinutes')) / opt('slotMinutes')
                                         )
                                     );
+                                    _ui.originalPosition.left += origColDragStart*colWidth;
+                                    // _ui.position.left += origColDragStart*colWidth;
+                                    // _ui.offset.left = _ui.position.left;
+                                    // _ui.helper.offset({left: ui.position.left});
+                                    // eventElement.draggable('option', 'cursorAt', {left:colContentLeft(0)});
+                                    // eventElement.draggable('option', 'refreshPositions', true);
                                     eventElement.draggable('option', 'grid', [colWidth, 1]);
                                     allDay = false;
                                 }
@@ -803,6 +816,19 @@ function AgendaEventRenderer() {
                     eventElement.draggable('option', 'revert', revert);
                 }, ev, 'drag');
             },
+            drag: function(ev, ui) {
+                // if (!allDay) {
+                //     // ui.position.left = getBodyContent().offset().left + colContentLeft(0) + origColDragStart*colWidth;
+                //     ui.position.left += origColDragStart*colWidth;
+                // }
+                // minuteDelta = Math.round((ev.pageY - originalTopPosition) / slotHeight) * opt('slotMinutes');
+                // if (minuteDelta != prevMinuteDelta) {
+                //     if (!allDay) {
+                //         updateTimeText(minuteDelta);
+                //     }
+                //     prevMinuteDelta = minuteDelta;
+                // }
+            },
             stop: function(ev, ui) {
                 hoverListener.stop();
                 clearOverlays();
@@ -826,6 +852,7 @@ function AgendaEventRenderer() {
                 eventElement.css('display', ''); // show() was causing display=inline
                 trigger('eventDragStop', eventElement, event, ev, ui);
                 isDragging = false;
+                origColDragStart = null;
                 //setOverflowHidden(false);
             }
         });
